File: <100/033/033.py
from fractions import Fraction
from math import prod

def pseudo_reduction(pair: tuple[int, int]) -> Fraction:
    (ns, ds) = (str(pair[0]), str(pair[1]))
    if ns[0] == ds[0]:
        return Fraction(f"{ns[1]}/{ds[1]}") if ds[1] != '0' else Fraction(101, 100)
    if ns[0] == ds[1]:
        return Fraction(f"{ns[1]}/{ds[0]}") if ds[0] != '0' else Fraction(101, 100)
    if ns[1] == ds[0]:
        return Fraction(f"{ns[0]}/{ds[1]}") if ds[1] != '0' else Fraction(101, 100)
    if ns[1] == ds[1]:
        return Fraction(f"{ns[0]}/{ds[0]}") if ds[0] != '0' else Fraction(101, 100)
    
    return Fraction(101, 100)


def is_ok(pair: tuple[int, int]) -> bool:
    proper_frac = Fraction(*pair)
    pseudo_frac = pseudo_reduction(pair)
    return proper_frac == pseudo_frac


def main() -> int:
    n_d_pairs = [(n, d) for d in range(11, 100) for n in range(11, d) if d%10 != 0 or n%10 != 0]
    good_pairs = filter(is_ok, n_d_pairs)
    in_frac_form = map(lambda pair: Fraction(*pair), good_pairs)
    prod_frac = prod(in_frac_form)
    return prod_frac.denominator


if __name__ == "__main__":
    answer = main()
    print(answer)

File: <100/033/033.jl
function pseudo_reduction(n :: Integer, d :: Integer) :: Rational{Integer}
  (ns, ds) = (string(n), string(d))
  if ns[1] == ds[1]
    return ns[2] == 0 || ds[2] == 0 ? 101 // 100 : parse(Integer, ns[2]) // parse(Integer, ds[2])
  end
  if ns[1] == ds[2]
    return ns[2] == 0 || ds[1] == 0 ? 101 // 100 : parse(Integer, ns[2]) // parse(Integer, ds[1])
  end
  if ns[2] == ds[1]
    return ns[1] == 0 || ds[2] == 0 ? 101 // 100 : parse(Integer, ns[1]) // parse(Integer, ds[2])
  end
  if ns[2] == ds[2]
    return ns[1] == 0 || ds[1] == 0 ? 101 // 100 : parse(Integer, ns[1]) // parse(Integer, ds[1])
  end

  return 101 // 100
end


function is_ok((n, d) :: Tuple{Integer, Integer}) :: Bool
  proper = n // d
  pseudo = pseudo_reduction(n, d)
  
  return proper == pseudo
end


function main() :: Nothing
  candidates :: Vector{Tuple{Int, Int}} = [(n, d) for d ∈ 11:99 for n ∈ 11:(d-1) if n%10 ≠ 0 || d%10 ≠ 0]
  good_pairs = filter(is_ok, candidates)
  rationals = map(t -> t[1] // t[2], good_pairs)
  rationals |> prod |> denominator |> println
end


@time main()
